src/evaluate/module.py

```diff
@@ -685,11 +686,19 @@ def _enforce_nested_string_type(self, schema, obj):
                 if isinstance(obj, (list, tuple)):
                     # obj is a list of dict
                     for k, dict_tuples in zip_dict(schema.feature, *obj):
-                        return [self._enforce_nested_string_type(dict_tuples[0], o) for o in dict_tuples[1:]]
+                        for sub_obj in dict_tuples[1:]:
+                            if sub_obj is not None:
```

Comment:
You also have to check if it's non empty here
```suggestion
                            if _check_non_null_non_empty_recursive(sub_obj, dict_tuples[0]):
```